coc/http.py

```diff
@@ -67,71 +68,62 @@ def __init__(self, method, path, kwargs: dict=None, api_page=False):
 
 
 class HTTPClient:
-    def __init__(self, client, loop, tokens=None, *, email, password, update_tokens=False):
+    def __init__(self, client, loop, email, password,
+                 key_names, key_count):
         self.client = client
         self.loop = loop
-        self.__session = None
+        self.__session = aiohttp.ClientSession(loop=self.loop)
         self.email = email
         self.password = password
-        self.update_tokens = update_tokens
+        self.key_names = key_names
+        self.key_count = key_count
 
-        asyncio.ensure_future(self.login(tokens))
+        loop.run_until_complete(self.get_keys())
 
-    async def login(self, tokens):
-        self.__session = aiohttp.ClientSession(loop=self.loop)
-
-        if not tokens:
-            ip = await self.get_ip()
-
-            response_dict, session = await self.login_to_site(self.email, self.password)
-            cookies = self.create_cookies(response_dict, session)
-            current_tokens = (await self.find_site_tokens(cookies))['keys']
-
-            self._tokens = [token['key'] for token in current_tokens if ip in token['cidrRanges']]
-            if not len(self._tokens):
-                if len(current_tokens) < 10:
-                    raise RuntimeError("No tokens found related to this IP. If you are interested in seeing this TODO"
-                                       " fixed, please open a PR on the repo.")
-                    # TODO: Create token
-                else:
-                    raise RuntimeError("There are already 10 tokens and none relate to this IP")
-        else:
-            self._tokens = tokens
-
-        self.tokens = cycle(self._tokens)
-        self.token = self._tokens[0]
+    async def get_keys(self):
+        key_count = self.key_count
+        ip = await self.get_ip()
+        response_dict, session = await self.login_to_site(self.email, self.password)
+        cookies = self.create_cookies(response_dict, session)
+        current_keys = (await self.find_site_keys(cookies))['keys']
+        self._keys = [key['key'] for key in current_keys if key['name'] == self.key_names]
+        available_keys = KEY_MAXIMUM - len(current_keys)
+
+        if len(self._keys) <= key_count:
+            keys_needed = key_count - len(self._keys)
+            if available_keys >= keys_needed:
+                for _ in range(keys_needed):
+                    key_description = "Created on {}".format(datetime.now().strftime('%c'))
+                    self._keys.append(await self.create_key(
+                        cookies,self.key_names, key_description, [ip]))
+            else:
+                await self.__session.close()
```

Comment:
might be better to call `close()` in case we want to pass anything else there in future, and it also checks if session is actually active (errors if not?)